technic/machines/register/battery_box.lua

```diff
@@ -188,7 +188,7 @@ function technic.register_battery_box(data)
 			"listring[context;upgrade2]"..
 			"listring[current_player;main]"
 	end
-
+	data.modname = data.modname or minetest.get_current_modname()
```

Comment:
Maybe instead of updating `data` table contents here use local variable 
```
local modname = data.modname or minetest.get_current_modname()
```
and similar with other files unless there's some good reason to actually update data table.

Reply:
Well it doesn't create another variable, and it's a bit clearer that the modname is part of the function arguments...

Reply:
> Well it doesn't create another variable

IMO it would be better to create another variable instead of modifying table

> and it's a bit clearer that the modname is part of the function arguments

That would not change at all or would it? If it would change then why would it change?

Only thing change I suggested affects is to not modify data table supplied from outside so that data table in hands of caller would not suddenly contain more data after using it with some register function.

Reply:
> Only thing change I suggested affects is to not modify data table supplied from outside so that data table in hands of caller would not suddenly contain more data after using it with some register function.

That makes sense 👍